certchain/certmanager/helpers_test.go

```diff
@@ -93,15 +94,31 @@ func (p *stubProducer) Stop() { close(p.out) }
 var _ certmanager.Cache = (*stubCache)(nil)
 
 type stubCache struct {
-	OnWrite chan *certchain.AugmentedChain
+	avail    chan struct{}
+	chainMap map[string]*certchain.AugmentedChain
 }
 
 func newStubCache() *stubCache {
-	return &stubCache{make(chan *certchain.AugmentedChain, 1)}
+	return &stubCache{
+		make(chan struct{}, 1),
+		make(map[string]*certchain.AugmentedChain),
+	}
+}
+
+func (c *stubCache) Read(digest string) (*certchain.AugmentedChain, error) {
+	<-c.avail
+	if _, ok := c.chainMap[digest]; !ok {
+		return nil, certmanager.ErrNotFound
+	}
+	return c.chainMap[digest], nil
 }
 
 func (c *stubCache) Write(ac *certchain.AugmentedChain) error {
-	c.OnWrite <- ac
+	if ac == nil {
```

Comment:
I think it's rare to verify explicitly that the argument is non-nil, maybe except for Contexts, although I have no idea about whether it's a good practice or a bad practice. It's presumably more common to let the program crash by nil object access.

Reply:
Removed